### DISCUSSION TITLE
components refreshing/reloading on route change (next/link) even when in root layout.js

### DISCUSSION THREAD

**Discussion by kizeesmack** [vercel/community#1796] [Help] [answered]
### Summary

Hello I am having a hard time understanding the persistent state of components in layout.js.

when developing and serving locally the component persists its rendered state. However when I deploy and use a vercel.app it reloads the canvas (three.js)

I am wondering if there is a reason for this difference in behaviour locally and in prod. 

https://arcade-ljcq.vercel.app/projects
I know its re-rendering because the 3d model component at the back resets.

**Comment by aleksrutins** [comment, marked as answer]
Components in the `app` directory (including `layout.js`) are server components by default, so navigation in production involves full-page reloads. I'm not entirely sure, but I think that the development behavior of client-side navigation is in order to have HMR work. You should be able to add `"use client"` to the top of your `layout.js` to prevent that, but that could have performance implications. If it works, a better solution (see [Moving Client Components to the Leaves](https://beta.nextjs.org/docs/rendering/server-and-client-components#moving-client-components-to-the-leaves)) would be to create a new client component with your Three.js code.
